app/models/concerns/article.rb

```diff
@@ -31,6 +35,15 @@ def set_user
     self.user = User.current unless User.current.nil?
   end
 
+  def set_channel
+    return if self.channel.present?
+    if self.user.is_a?(BotUser)
+      self.channel = "api"
+    else
+      self.channel = "manual"
+    end
```

Comment:
recommended to use conditional operator
i.e self.channel = self.user.is_a?(BotUser) ?  'api'  :  'manual'